homeassistant/components/smartthings/sensor.py

```diff
@@ -360,3 +365,47 @@ def state(self):
             return three_axis[self._index]
         except (TypeError, IndexError):
             return None
+
+
+class SmartThingsPowerConsumptionReportSensor(SmartThingsEntity):
+    """Define a SmartThings Power Consumption Report."""
+
+    def __init__(self, device):
+        """Init the class."""
+        super().__init__(device)
+        self._name = "Power Consumption Report"
+
+    @property
+    def name(self) -> str:
+        """Return the name of the binary sensor."""
+        return f"{self._device.label} {self._name}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return f"{self._device.device_id}.powerConsumptionReport"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._device.status.power_consumption_energy
+
+    @property
+    def device_class(self) -> str:
+        """Return the device class of the sensor."""
+        return DEVICE_CLASS_ENERGY
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return the unit this state is expressed in."""
+        return ENERGY_WATT_HOUR
+
+    @property
+    def device_state_attributes(self):
+        """Return attributes of the sensor."""
+        state_attributes = {}
+        for attribute, short_name in POWER_CONSUMPTION_REPORT_NAMES.items():
```

Comment:
An alternative to making these measurements state attributes is to make them separate sensor entities and have them be disabled by default. The user can then enable one or more if they want.

Reply:
@andrewsayre and I discussed a bit - the trouble is that the spec is not uniform so some of the sensor entities may not get any data.  With the state attributes at least we can provide the user with exactly the attributes that will be receiving data.  Then someone can use template sensor if they want to break it out.

> I don't think we should represent each sub-attribute as a discrete sensor instance, as they are not guaranteed to be present, unlike three-axsis. Instead, I think it should be a single sensor instance that reports `energy` as the primary value and then optionally adds the other values as state attributes ONLY if they are present.

Reply:
May this lack of data be varying over time for the user? If not we can check the data before adding the entity.

Reply:
I like that idea - adding only the sensors that have data at HA startup / device setup.  I will wait for feedback from @andrewsayre .  I think that variance over time is a possibility as the energy reports get populated quite infrequently.